### PR TITLE
RUBY-2630 Implement load balanced topology type


### PR DESCRIPTION
**Pull request by p-mongo**

### CODE REVIEW

**Comment by p-mongo** [comment]
https://jira.mongodb.org/browse/RUBY-2630

File: lib/mongo/cluster/topology/load_balanced.rb [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+      # Defines behavior for when a cluster is in load-balanced topology.
+      class LoadBalanced < Base
+
+        # The display name for the topology.
+        NAME = 'LoadBalanced'.freeze
+
+        # Get the display name.
+        #
+        # @example Get the display name.
```
Comment (comandeo) [inline lib/mongo/cluster/topology/load_balanced.rb:30]: I would rather remove this example while usage of this method is rather clear. But if we want to keep it, let's change to `LoadBalanced.display_name`.

File: lib/mongo/server/description.rb [hunk 2]
```diff
@@ -237,6 +243,10 @@ def initialize(address, config = {}, average_round_trip_time = nil)
       # @return [ Hash ] The actual result from the ismaster command.
       attr_reader :config
 
+      def load_balancer?
```
Comment (comandeo) [inline lib/mongo/server/description.rb:246]: Maybe a short docstring here? Since all other methods in this class have one.

**Review by comandeo — APPROVED** [review]

File: lib/mongo/cluster/topology/load_balanced.rb (hunk 1 above)
Comment (p-mongo) [reply lib/mongo/cluster/topology/load_balanced.rb:30]: I removed all of the examples.

File: lib/mongo/server/description.rb (hunk 2 above)
Comment (p-mongo) [reply lib/mongo/server/description.rb:246]: Added, there should've been one here.

**Pull request merged**